Gems/ROS2/Code/Source/Camera/PostProcessing/ROS2ImageEncodingConversionComponent.cpp

```diff
@@ -0,0 +1,206 @@
+            { "mono8", ImageEncoding::Mono8 },
+            { "mono16", ImageEncoding::Mono16 },
+        };
+
+        void Rgba8ToRgb8(sensor_msgs::msg::Image& image)
+        {
+            const std::string inputEncoding = ImageEncodingNames.at(ImageEncoding::RGBA8);
+            const std::string outputEncoding = ImageEncodingNames.at(ImageEncoding::RGB8);
+            AZ_Assert(image.encoding == inputEncoding, "Image encoding is %s, expected %s", image.encoding.c_str(), inputEncoding.c_str());
+            AZ_Assert(image.step == image.width * 4, "Image step is not width * 4");
+            AZ_Assert(image.data.size() == image.step * image.height, "Image data size is not step * height");
+
+            // Perform conversion in place
+            for (size_t pixelId = 0; pixelId < image.width * image.height; ++pixelId)
+            {
+                size_t pixelOffsetIn = pixelId * 4;
+                size_t pixelOffsetOut = pixelId * 3;
+                image.data[pixelOffsetOut] = image.data[pixelOffsetIn];
+                image.data[pixelOffsetOut + 1] = image.data[pixelOffsetIn + 1];
+                image.data[pixelOffsetOut + 2] = image.data[pixelOffsetIn + 2];
+            }
+            image.encoding = outputEncoding;
+            image.step = image.width * 3;
+            image.data.resize(image.step * image.height);
+        }
+
+        const AZStd::unordered_map<EncodingConversion, const AZStd::function<void(sensor_msgs::msg::Image&)>> supportedFormatChange = {
+            { { ImageEncoding::RGBA8, ImageEncoding::RGB8 }, Rgba8ToRgb8 },
+        };
+
+        AZ::Outcome<void, AZStd::string> ValidateEncodingConversion(EncodingConversion newConversion)
+        {
+            if (newConversion.encodingIn == newConversion.encodingOut)
+            {
+                return AZ::Failure(AZStd::string("Conversion to same type if forbidden"));
+            }
+            if (supportedFormatChange.find(newConversion) == supportedFormatChange.end())
+            {
+                return AZ::Failure(AZStd::string::format(
+                    "Unsupported encoding change from %s to %s",
+                    ImageEncodingNames.at(newConversion.encodingIn),
+                    ImageEncodingNames.at(newConversion.encodingOut)));
+            }
+            return AZ::Success();
+        }
+
+    } // namespace
+
+    void EncodingConversion::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<EncodingConversion>()
+                ->Version(0)
+                ->Field("EncodingIn", &EncodingConversion::encodingIn)
+                ->Field("EncodingOut", &EncodingConversion::encodingOut);
+
+            if (AZ::EditContext* ec = serializeContext->GetEditContext())
+            {
+                ec->Class<EncodingConversion>("Encoding Conversion", "Specifies encoding conversion")
+                    ->DataElement(
```

Comment:
Please correct me if I am wrong. Does the user need to put encoding of image coming from Atom?

Reply:
Yes, user needs to put encoding from Atom, this allows for having separate conversions  from
- rgba8
- mono8
to any other encoding if desired, It allows for piping encoding conversions with any other operations.